extensions/publora/src/schedule-post.tsx

```diff
@@ -0,0 +1,89 @@
+import { Form, ActionPanel, Action, LaunchProps, showToast, getPreferenceValues } from "@raycast/api";
+import { useCachedState } from "@raycast/utils";
+import { useState, useEffect } from "react";
+
+import { Post, Platform } from "./types";
+import { fetchPlatforms, schedulePost } from "./api";
+
+export default function Command(props: LaunchProps<{ draftValues: Post }>) {
+  async function handleSubmit(post: Post) {
+    try {
+      await schedulePost(api_key, post);
+      showToast({ title: "Submitted form", message: "Your post scheduled successfully" });
+      setContent("");
+      setDue(null);
+      setPlatforms([]);
+    } catch (error) {
+      console.error("Error scheduling post:", error);
+      showToast({ title: "Error", message: "Failed to schedule post" });
+    }
+  }
+
+  const api_key = getPreferenceValues<{ api_key: string }>().api_key;
```

Comment:
There is an `ExtensionPreferences` in the global scope. You can simply pass in the `ExtensionPreferences` its generic type.

```suggestion
  const api_key = getPreferenceValues<ExtensionPreferences>().api_key;
```

Reply:
Fixed